### PR TITLE
fix(seo): preserve partial response integrity

### PR DESCRIPTION
**Pull request by blakeox**
## What changed
- apply query robots metadata rewriting to non-5xx HTML error responses
- skip rewriting `206 Partial Content` responses so byte ranges and entity headers remain valid
- add focused tests for 404 HTML and partial HTML behavior

## Why
The query URL policy must cover HTML error responses, but rewriting a partial response would change its body while retaining range semantics. This closes both edge cases without changing the normal response contract.

## Validation
- focused edge tests: 17 passed
- Prettier check passed
- `pnpm typecheck` passed: 351 files, 0 diagnostics
- pre-push typecheck and lint passed

---

> [!NOTE]
> **Low Risk**
> Narrow edge HTML rewrite tweak for crawl metadata and range responses; no auth, data, or cache-contract changes.
> 
> **Overview**
> Query-bearing HTML now gets the noindex robots meta on **non-5xx origin errors** (e.g. 404), not only successful and 5xx fallback paths.
> 
> `addQueryNoindexMeta` **skips `206 Partial Content`**, so range bodies and entity headers are not rewritten. Tests cover 404 query HTML and the 206 skip.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c890935e250dcb196441978dbc6c388dc6bbcf13. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: functions/routes/assets.ts [hunk 1]
```diff
@@ -10,7 +10,13 @@ const ROBOTS_META_PATTERN = /<meta\b(?=[^>]*\bname=["']robots["'])[^>]*>/i;
 /** Keep query-bearing HTML crawl policy consistent in both headers and markup. */
 export async function addQueryNoindexMeta(response: Response, url: URL): Promise<Response> {
   const contentType = response.headers.get('content-type')?.toLowerCase() ?? '';
-  if (!url.search || !contentType.includes('text/html') || !response.body) return response;
+  if (
+    !url.search ||
+    response.status === 206 ||
```
[the remark this answers is not in the document]
Comment (blakeox) [reply functions/routes/assets.ts:15]: Resolved in commit 209e15b1. Query-bearing 206 responses now bypass both robots-meta rewriting and downstream theme personalization, preserving partial body and range/entity semantics; the regression test is included.
